modules/caddyhttp/encode/encode.go

```diff
@@ -75,11 +80,49 @@ func (enc *Encode) Provision(ctx caddy.Context) error {
 	if enc.MinLength == 0 {
 		enc.MinLength = defaultMinLength
 	}
+
+	if len(enc.Types) == 0 {
+		enc.Types = []string{"*"} // backwards compatibility
+
+		// sane default for text-based content ?
```

Comment:
We like sane defaults

Reply:
with the implementation via `ResponseMatcher` the default is:
```
header Content-Type text/*
header Content-Type application/json
header Content-Type application/javascript
header Content-Type application/xhtml+xml
header Content-Type application/atom+xml
header Content-Type application/rss+xml
header Content-Type image/svg+xml
```

Reply:
Changed the default to:

```
header Content-Type text/*
header Content-Type application/json*
header Content-Type application/javascript*
header Content-Type application/xhtml+xml*
header Content-Type application/atom+xml*
header Content-Type application/rss+xml*
header Content-Type image/svg+xml*
```

to account for any `;charset=xxx` that may be appended to the header.

Reply:
Are there any MIME types that could match these blobs except for the `;charset=...` addition?
Could we use a regexp instead (or is it too much of a performance hit, maybe?)

Reply:
Regexps aren't supported by the `ResponseMatcher` type, yet. It would be nice the have a `header_regexp` directive from `RequestMatcher`s as well, but I think this would be out of the scope of this PR. Maybe an additional one?

Reply:
I agree we should have a header_regexp matcher. If you want to do a follow-up with that, that'd be great 👍
I am a bit worried that it would be detrimental to performance for the `encode` defaults though, so a bit of benchmarking may be beneficial to see if it makes any appreciable difference.